File: src/syntax/lesson21_practice/Practice.java
package syntax.lesson21_practice;

public class Practice {
    public static void main(String[] args) {

        System.out.println("The winner is " + getWinner(230, 220, 190, 200));

    }

    public static String getWinner(int gryffindorPoints, int slytherinPoints, int ravenclawPoints, int hufflepuffPoints) {

        if (gryffindorPoints > slytherinPoints && gryffindorPoints > ravenclawPoints && gryffindorPoints > hufflepuffPoints) {
            return "Gryffindor!";
        } else if (slytherinPoints > gryffindorPoints && slytherinPoints > ravenclawPoints && slytherinPoints > hufflepuffPoints) {
            return "Slytherin!";
        } else if (ravenclawPoints > gryffindorPoints && ravenclawPoints > slytherinPoints && ravenclawPoints > hufflepuffPoints) {
            return "Ravenclaw!";
        } else {
            return "Hufflepuff!";
        }
    }
}

File: src/syntax/lesson21_practice/Practice.kt
package syntax.lesson21_practice

fun main() {

    println("The winner is ${getWinner(300, 320, 280, 310)}")

}

fun getWinner(
    gryffindorPoints: Int, slytherinPoints: Int,
    ravenclawPoints: Int, hufflepuffPoints: Int
): String {

    return if (gryffindorPoints > slytherinPoints
        && gryffindorPoints > ravenclawPoints
        && gryffindorPoints > hufflepuffPoints
    ) {
        "Gryffindor!"
    } else if (slytherinPoints > gryffindorPoints
        && slytherinPoints > ravenclawPoints
        && slytherinPoints > hufflepuffPoints
    ) {
        "Slytherin!"
    } else if (ravenclawPoints > gryffindorPoints
        && ravenclawPoints > slytherinPoints
        && ravenclawPoints > hufflepuffPoints
    ) {
        "Ranerclaw!"
    } else {
        "Hufflepuff!"
    }
}